include/libdnf/common/format.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_COMMON_FORMAT_HPP
+#define LIBDNF_COMMON_FORMAT_HPP
+
+#include <fmt/format.h>
+
+
+namespace libdnf {
+
+/// Format `args` according to the `runtime_format_string`, and return the result as a string.
+/// Unlike C++20 `std::format`, the format string of the `libdnf::sformat` function template
+/// does not have to be a constant expression.
+template <typename... Args>
+inline std::string sformat(std::string_view runtime_format_string, Args&&... args) {
```

Comment:
Do we name this sformat or just format?

I think the main consideration might be it'll be on libdnf's interface. I'm not sure the s at the beginning adds anything?

I'm also not sure whether we shouldn't put it into the libdnf::utils namespace? I don't have a strong opinion either way though.

@j-mracek @m-blaha wanna cast a vote?

The PR looks good otherwise.

Reply:
I also don't have a strong opinion.